web/src/narrow.js

```diff
@@ -425,8 +425,13 @@ export function activate(raw_terms, opts) {
         let msg_list;
         let restore_rendered_list = false;
         for (const list of message_lists.all_rendered_message_lists()) {
-            if (is_combined_feed_global_view && list.preserve_rendered_state) {
-                assert(list.data.filter.is_in_home());
+            if (is_combined_feed_global_view && list.data.filter.is_in_home()) {
+                if (opts.then_select_id > 0 && !list.get(opts.then_select_id)) {
+                    // We don't have the target message in the current rendered list.
```

Comment:
Do you think it's most correct the check if the message is in the list, or just if it's in the range between the start/end of the list? This impacts what happens if we try to anchor on a message ID that is in the fetched range but doesn't exist anymore (say, because of some bug where the message was deleted/moved).

Reply:
Yeah, makes sense to do that check instead. Changed!